pull.go

```diff
@@ -141,9 +146,53 @@ func localImageNameForReference(ctx context.Context, store storage.Store, srcRef
 }
 
 // Pull copies the contents of the image from somewhere else to local storage.
-func Pull(ctx context.Context, imageName string, options PullOptions) (types.ImageReference, error) {
+func Pull(ctx context.Context, imageName string, options PullOptions) error {
+	spec := imageName
 	systemContext := getSystemContext(options.SystemContext, options.SignaturePolicyPath)
-	return pullImage(ctx, options.Store, imageName, options, systemContext)
+	srcRef, err := alltransports.ParseImageName(spec)
+	if err != nil {
+		if options.Transport == "" {
+			options.Transport = util.DefaultTransport
+		}
+		logrus.Debugf("error parsing image name %q, trying with transport %q: %v", spec, options.Transport, err)
+		transport := options.Transport
+		if transport != util.DefaultTransport {
+			transport = transport + ":"
+		}
+		spec = transport + spec
+		srcRef2, err2 := alltransports.ParseImageName(spec)
+		if err2 != nil {
+			return errors.Wrapf(err2, "error parsing image name %q", spec)
+		}
+		srcRef = srcRef2
+	}
+	var names []string
+	if options.AllTags {
+		tags, err := docker.GetRepositoryTags(ctx, systemContext, srcRef)
+		if err != nil {
+			return errors.Wrapf(err, "error geting repository tags")
+		}
+		for _, tag := range tags {
+			name := spec + ":" + tag
+			names = append(names, name)
+		}
+	} else {
+		names = append(names, spec)
+	}
+	for _, name := range names {
+		fmt.Printf("Pulling %s\n", name)
+		ref, err := pullImage(ctx, options.Store, name, options, systemContext)
+		if err != nil {
+			return errors.Wrapf(err, "error pulling image %q", name)
+		}
+		img, err := is.Transport.GetStoreImage(options.Store, ref)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("%s\n", img.ID)
```

Comment:
Does PullImage do this for us?

Reply:
Because the previous [cmd/buildah/pull.go](https://github.com/containers/buildah/pull/1263/files#diff-4ca994502fc61e8b4456e9a91bcef926L124) did the same, I just copied this code. I think the output image ID of the pull down is more helpful to the user.

Reply:
ok